DQM/src/DQM/VisiblesFeatureProducer.cxx

```diff
@@ -0,0 +1,328 @@
+#include "DQM/VisiblesFeatureProducer.h"
+
+// LDMX
+#include "DetDescr/HcalID.h"
+#include "DetDescr/SimSpecialID.h"
+#include "Ecal/Event/EcalHit.h"
+#include "Hcal/Event/HcalHit.h"
+#include "SimCore/Event/SimCalorimeterHit.h"
+#include "SimCore/Event/SimParticle.h"
+#include "SimCore/Event/SimTrackerHit.h"
+#include "Tracking/Event/Track.h"
+
+// C++
+#include <algorithm>
+#include <cmath>
+#include <fstream>
+#include <iostream>
+#include <numeric>
+
+namespace dqm {
+
+void VisiblesFeatureProducer::configure(
+    framework::config::Parameters &parameters) {
+  training_ = parameters.get<bool>("training");
+  training_file_ = parameters.get<std::string>("training_file");
+
+  beam_energy_mev_ = parameters.get<double>("beam_energy");
+
+  // collection names
+  hcal_rec_collection_ = parameters.get<std::string>("hcal_rec_coll_name");
+  hcal_rec_pass_name_ = parameters.get<std::string>("hcal_rec_pass_name");
+
+  ecal_rec_collection_ = parameters.get<std::string>("ecal_rec_coll_name");
+  ecal_rec_pass_name_ = parameters.get<std::string>("ecal_rec_pass_name");
+
+  recoil_from_tracking_ = parameters.get<bool>("recoil_from_tracking");
+  track_collection_ = parameters.get<std::string>("track_collection");
+  track_pass_name_ = parameters.get<std::string>("track_pass_name");
+
+  sp_collection_ = parameters.get<std::string>("sp_coll_name");
+  sp_pass_name_ = parameters.get<std::string>("sp_pass_name");
+  sim_particles_pass_name_ =
+      parameters.get<std::string>("sim_particles_pass_name");
+}
+
+bool VisiblesFeatureProducer::inList(std::vector<int> parents, int track_id) {
+  bool in_list = false;
+  for (const int &i : parents) {
+    if (i == track_id) {
+      in_list = true;
+    }
+  }
+  return in_list;
+}
+
+void VisiblesFeatureProducer::analyze(const framework::Event &event) {
+  std::vector<double> bdt_features;
+
+  auto particle_map{event.getMap<int, ldmx::SimParticle>(
```

Comment:
Same comment about getting the maps/collections as the other file (I'm just too lazy to put in suggestions).